server/src/test/java/org/elasticsearch/test/TestUtils.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test;
+
+import org.elasticsearch.core.Nullable;
+
+import java.util.concurrent.Callable;
+
+/**
+ * Test utilities
+ */
+public class TestUtils {
+    public static <T> T assetNoException(Callable<T> callable, @Nullable String msg) {
+        try {
+            return callable.call();
+        } catch (Throwable th) {
+            throw new AssertionError(((msg != null) ? msg + "; original error: " : "unexpected error: ") + th.getMessage());
+        }
+    }
+}
```

Comment:
There are benefits with proper assertions, rather than letting a test just fail. Mainly, the ability to print a more indicative message on what may be the actual regression leading to future failures. In addition, anything that expects assertion failures to be the same would benefit from that. For example, in IntelliJ an assertion failure will have a different icon from any other uncaught exception, which I think is preferable for troubleshooting. This was a very simple attempt to offer a functionality similar to `assertDoesNotThrow()` that was introduced in JUnit 5.

> we try to be very conservative with utility classes since we have so many

No problem, removing